crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -477,7 +474,7 @@ impl Timeline {
     pub async fn edit(
         &self,
         item: &EventTimelineItem,
-        new_content: RoomMessageEventContentWithoutRelation,
+        new_content: EditedContent,
```

Comment:
I do realize this function is missing some checks, and that predates your PR: what if we're trying to edit a poll with a room message? We'd need to check that the item content kind matches the `EditedContent` variant too.

Can you add a test for this please? :pleading_face:

Reply:
That's already done [for edits to remote events](https://github.com/matrix-org/matrix-rust-sdk/blob/58ab52dc5f9853974658026bb4ffc70c922b2d43/crates/matrix-sdk/src/room/edit.rs#L158), although I'm not sure it's tested. Should I do that too for local echoes?

Reply:
Ah, you're right that the `make_edit` function a few layers below will fail if the timeline item doesn't contain something matching the event type. Yes, having something like this for 1. polls at the `make_edit` function level (I think it was there, IIRC), 2. local echoes for poll edits, will be useful.

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/3951/commits/26c2da4dd6cd967f66afa599b43b58866bd934db should handle this.